gensim/corpora/textcorpus.py

```diff
@@ -305,7 +302,7 @@ def sample_texts(self, n, seed=None, length=None):
             # This means that length was set to be greater than number of items in corpus
             # and we were not able to sample enough documents before the stream ended.
             raise ValueError(
-                "length {0:d} greater than number of documents in corpus {1:d}".format(length, seen_documents))
+                "length {0:d} greater than number of documents in corpus {1:d}".format(length, i + 1))
```

Comment:
This will fail with an exception if there are no texts (`i` will be uninitialized).